### PR TITLE
use ncdf:: for all functions

### PR DESCRIPTION
**Pull request by robkooper**
use ncdf:: for all functions as well as a fix to have correct lat/lon descriptions

lat is north/south so unit is degrees_north
lon is east/west so unit is degrees_east

### CODE REVIEW

File: models/ed/R/model2netcdf.ED2.R [hunk 1]
```diff
@@ -562,14 +562,14 @@ model2netcdf.ED2 <- function(outdir, sitelat, sitelon, start_date, end_date) {
     var[[46]] <- mstmipvar("Qsb", lat, lon, t, zg)
     
     ## write ALMA
-    nc <- nc_create(file.path(outdir, paste(yrs[y], "nc", sep = ".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(yrs[y], "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(yrs[y], "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       ncvar_put(nc, var[[i]], out[[i]])
```
Comment (tonygardella) [inline models/ed/R/model2netcdf.ED2.R]: Does this need to be ```ncdf4::ncvar_put(nc, var[[i]], out[[i]])```

File: models/fates/R/model2netcdf.FATES.R [hunk 2]
```diff
@@ -35,13 +35,13 @@ model2netcdf.FATES <- function(outdir) {
         oname <- file.path(dirname(fname), paste0(year, ".nc"))
         logger.info(paste("model2netcdf.FATES:", fname, "to", oname))
         file.copy(fname, oname)
-        nc <- nc_open(oname, write = TRUE)
+        nc <- ncdf4::nc_open(oname, write = TRUE)
         
         ## FATES time is in multiple columns, create 'time'
         day  <- ncvar_get(nc, "mdcur")  #current day (from base day)
```
Comment (tonygardella) [inline models/fates/R/model2netcdf.FATES.R]: ```ncdf4::ncvar_get``` here and one line below

File: models/gday/R/model2netcdf.GDAY.R [hunk 3]
```diff
@@ -103,25 +103,25 @@ model2netcdf.GDAY <- function(outdir, sitelat, sitelon, start_date, end_date) {
...
     # 'kgC/m2', list(lon,lat,t), -999)
     
     # ******************** Declare netCDF variables ********************#
     
     ### Output netCDF data
-    nc <- nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       ncvar_put(nc, var[[i]], output[[i]])
```
Comment (tonygardella) [inline models/gday/R/model2netcdf.GDAY.R]: ```ncdf4::ncvar_put``` here

File: models/jules/R/model2netcdf.JULES.R [hunk 4]
```diff
@@ -32,6 +32,6 @@ model2netcdf.JULES <- function(outdir) {
     ## JULES time is in seconds; convert to DOY
     time <- ncvar_get(nc, "time") / 86400
     ncvar_put(nc, "time", time)
```
Comment (tonygardella) [inline models/jules/R/model2netcdf.JULES.R]: ```ncd4::ncvar_put``` here

File: models/linkages/R/model2netcdf.LINKAGES.R [hunk 5]
```diff
@@ -99,34 +99,34 @@ model2netcdf.LINKAGES <- function(outdir, sitelat, sitelon, start_date=NULL, end
...
     #******************** Declar netCDF variables ********************#
 
 
     ### Output netCDF data
-    nc <- nc_create(file.path(outdir, paste(years[y],"nc", sep=".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(years[y],"nc", sep=".")), var)
     varfile <- file(file.path(outdir, paste(years[y], "nc", "var", sep=".")), "w")
     for(i in 1:length(var)){
       print(i)
       ncvar_put(nc,var[[i]],output[[i]])
```
Comment (tonygardella) [inline models/linkages/R/model2netcdf.LINKAGES.R]: ```ncdf4::ncvar_put``` here

File: models/lpjguess/R/model2netcdf.LPJGUESS.R [hunk 6]
```diff
@@ -113,15 +113,15 @@ model2netcdf.LPJGUESS <- function(outdir, sitelat, sitelon, start_date, end_date
     # ******************** Declare netCDF variables ********************#
     
     ### Output netCDF data
-    nc <- nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       # print(i)
       ncvar_put(nc, var[[i]], output[[i]])
```
Comment (tonygardella) [inline models/lpjguess/R/model2netcdf.LPJGUESS.R]: ```ncdf4::ncvar_put``` here

File: models/maat/R/model2netcdf.MAAT.R [hunk 7]
```diff
@@ -118,15 +118,15 @@ model2netcdf.MAAT <- function(outdir, sitelat=-999, sitelon=-999, start_date=NUL
     var[[4]]  <- mstmipvar("stomatal_conductance", lat, lon, t, NA)
     
     ### Output netCDF data
-    nc <- nc_create(file.path(outdir, paste(y,"nc", sep=".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(y,"nc", sep=".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep=".")), "w")
     for(i in 1:length(var)){
       print(i) # just on for debugging
       ncvar_put(nc,var[[i]],output[[i]])
```
Comment (tonygardella) [inline models/maat/R/model2netcdf.MAAT.R]: ```ncdf4::ncvar_put``` here

File: models/maespa/R/model2netcdf.MAESPA.R [hunk 8]
```diff
@@ -74,25 +74,25 @@ model2netcdf.MAESPA <- function(outdir, sitelat, sitelon, start_date, end_date,
...
+    var[[4]] <- ncdf4::ncvar_def("Qh", units = ("W m-2"), dim = list(lat, lon, t), missval = -999, longname = "Sensible Heat Flux")
+    var[[5]] <- ncdf4::ncvar_def("Qle", units = ("W m-2"), dim = list(lat, lon, t), missval = -999, longname = "latent Heat Flux")
     
     ### Output netCDF data
-    nc <- nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
+    nc <- ncdf4::nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       # print(i)
       ncvar_put(nc, var[[i]], output[[i]])
```
Comment (tonygardella) [inline models/maespa/R/model2netcdf.MAESPA.R]: ```ncdf4::ncvar_put``` here

File: models/biocro/R/model2netcdf.BIOCRO.R [hunk 9]
```diff
@@ -85,9 +85,9 @@ model2netcdf.BIOCRO <- function(result, genus = NULL, outdir, lat = -9999, lon =
     
     ncfile <- file.path(outdir, paste0(yeari, ".nc"))
     if(file.exists(ncfile)){
-      nc <- nc_open(ncfile, write = TRUE)
+      nc <- ncdf4::nc_open(ncfile, write = TRUE)
     } else {
-      nc <- nc_create(filename = file.path(outdir, paste0(yeari, ".nc")),
+      nc <- ncdf4::nc_create(filename = file.path(outdir, paste0(yeari, ".nc")),
                       vars = vars)      
       ncatt_put(nc, 0, "description",
```
Comment (tonygardella) [inline models/biocro/R/model2netcdf.BIOCRO.R]: ```ncdf4::ncatt_put```  here

**Comment by robkooper** [comment]
turns out this was done already in another pull request. declining.

**Comment by tonygardella** [comment]
@robkooper which pull request was this done in?

**Comment by tonygardella** [comment]
@robkooper nevermind I saw it in #1111. Saw some missing so i'll assign that to someone in #1116.

**Pull request closed without merging**